benches/sha256.rs

```diff
@@ -371,6 +365,166 @@ fn prove_compressed_benchmarks(c: &mut Criterion) {
+// fn sha256_nivc_prove_compressed<M: measurement::Measurement>(
+//     prove_params: ProveParams,
+//     c: &mut BenchmarkGroup<'_, M>,
+//     state: Rc<RefCell<State>>,
+// ) {
+//     let ProveParams {
+//         arity,
+//         n: _,
+//         reduction_count,
+//     } = prove_params;
+
+//     let limit = 10000;
+
+//     let store = &mut Store::<Fr>::new();
+//     let cproc_sym = user_sym(&format!("sha256_ivc_{arity}"));
+
+//     let lang = Lang::<Fr, Sha256Coproc<Fr>>::new_with_bindings(
+//         store,
+//         vec![(cproc_sym, Sha256Coprocessor::new(arity).into())],
+//     );
+//     let lang_rc = Arc::new(lang.clone());
+
+//     // use cached public params
+//     let instance = Instance::new_supernova(reduction_count, lang_rc.clone(), true);
+//     let pp = pp(&instance, Utf8Path::new(PUBLIC_PARAMS_PATH)).unwrap();
+
+//     c.bench_with_input(
+//         BenchmarkId::new(prove_params.name(), arity),
+//         &prove_params,
+//         |b, prove_params| {
+//             let env = empty_sym_env(store);
+//             let ptr = sha256_ivc(
+//                 store,
+//                 state.clone(),
+//                 black_box(prove_params.arity),
+//                 black_box(prove_params.n),
+//                 (0..prove_params.n).collect(),
+//             );
+
+//             let prover = SuperNovaProver::new(prove_params.reduction_count, lang.clone());
+
+//             let frames = &prover
+//                 .get_evaluation_frames(ptr, env, store, limit, lang_rc.clone())
+//                 .unwrap();
+
+//             b.iter_batched(
+//                 || (frames, lang_rc.clone()),
+//                 |(frames, lang_rc)| {
+//                     let (proof, _, _, _, _) = prover.prove(&pp, frames, store, lang_rc).unwrap();
+//                     let compressed_result = proof.compress(&pp).unwrap();
+
+//                     let _ = black_box(compressed_result);
+//                 },
+//                 BatchSize::LargeInput,
+//             )
+//         },
+//     );
+// }
+
+// TODO: Add when compressed SNARK is implemented for SuperNova
+// fn prove_compressed_benchmarks(c: &mut Criterion) {
+//     tracing::debug!("{:?}", &*lurk::config::CONFIG);
+//     let reduction_counts = vec![10, 100];
+//     let batch_sizes = vec![1, 2, 5, 10, 20];
+//     let mut group: BenchmarkGroup<'_, _> = c.benchmark_group("prove_compressed");
+//     group.sampling_mode(SamplingMode::Flat); // This can take a *while*
+//     group.sample_size(10);
+//     let state = State::init_lurk_state().rccell();
+
+//     for &n in batch_sizes.iter() {
+//         for &reduction_count in reduction_counts.iter() {
+//             let prove_params = ProveParams {
+//                 arity: 1,
+//                 n,
+//                 reduction_count,
+//             };
+//             sha256_nivc_prove_compressed(prove_params, &mut group, state.clone());
+//         }
+//     }
+// }
```

Comment:
What's the value of having this commented out given we have access to prove_compressed for IVC above?

Reply:
True